src/helpers.php

```diff
@@ -95,3 +96,49 @@ function get_locale_name($code, $locale = 'en', $capitalize = true)
         }
     }
 }
+
+if (! function_exists('trans_current_route')) {
+    /**
+     * Retrieve the current route in another locale.
+     *
+     * @param  string|null  $locale
+     * @param  string|null  $fallback
+     * @param  bool  $absolute
+     * @return string
+     */
+    function trans_current_route(string $locale = null, string $fallback = null, bool $absolute = true): string
+    {
+        if (is_null($fallback)) {
+            $fallback = url(request()->server('REQUEST_URI'));
+        }
+
+        $route = Route::getCurrentRoute();
+
+        if (! $route) {
+            return $fallback;
+        }
+
+        $name = Str::replaceFirst(
+            locale().'.',
+            "{$locale}.",
+            $route->getName()
+        );
+
+        if (! $name || ! in_array($locale, locales()) || ! Route::has($name)) {
+            return $fallback;
+        }
+
+        $current = locale();
+
+        locale($locale);
+
+        $url = route($name, array_merge(
+            (array) $route->parameters,
+            (array) request()->getQueryString()
+        ), $absolute);
+
+        locale($current);
```

Comment:
I don't think that this would cause a race condition— `locale()` is a helper for the App facade's `setLocale` method, which according to Laravel's docs "sets the locale for a specific request at runtime": https://laravel.com/docs/9.x/localization#configuring-the-locale

So it's not a global change for all user of the app, but is specific to each request. Still good to keep an eye out for this.